### DISCUSSION TITLE
How can I change polygon fill colors based in data table column?

### DISCUSSION THREAD

**Discussion by PatronMaster** [apache/superset#10437] [General]
Hi Experts,

I am using the deck.gl Polygon. 

![image](https://user-images.githubusercontent.com/7825849/88572458-13ab6600-d037-11ea-86f8-ee53459b4a67.png)

I used Javascript tooltip and onClick to get the data from my data table.

[image](https://user-images.githubusercontent.com/7825849/88572699-84528280-d037-11ea-8cc0-03ba29d4e194.png)

**My goal now it just change the polygon fill color.**

I have one column in data table that have the color of each polygon, like (#cccce0, #7f7fb2) and column for the legend (class of this polygon).

I tried to use javascript data interceptor (d[i].color) I can change the color column in data table but I cannot use to change polygon fill color.

The customize tab also do not give any option for that.

![image](https://user-images.githubusercontent.com/7825849/88574020-8ae1f980-d039-11ea-9336-2092cef3ecb4.png).

I just want do do the following:

- polygon 1, RED, means BAD so legend = BAD
- polygon 2, GREEN, means GOOD so legend = GOOD 
- polygon 3, RED, means BAD so legend = BAD
- polygon 4, RED, means BAD so legend = BAD

How can do that with Superset?

**Comment by VSZM** [comment]
Hi! Did you manage to get this working?

**Comment by PatronMaster** [comment]
NoEm 29/03/2021 21:57, Szegedi Gábor ***@***.***> escreveu:
Hi! Did you manage to get this working?
